File: convert_excel_to_csv.py
import pandas as pd
import os

# Path to the Excel file
excel_file = 'Other/visiondb.xlsx'

# Path for the output CSV file
csv_file = 'Other/visiondb.csv'

try:
    # Read the Excel file
    print(f"Reading Excel file: {excel_file}")
    df = pd.read_excel(excel_file)
    
    # Save as CSV
    print(f"Converting to CSV: {csv_file}")
    df.to_csv(csv_file, index=False)
    
    print(f"Conversion complete! CSV file saved to: {csv_file}")
    print(f"Number of rows: {len(df)}")
    print(f"Columns: {', '.join(df.columns)}")
    
except Exception as e:
    print(f"Error converting file: {str(e)}")

File: convert_excel_to_csv.js
const xlsx = require('xlsx');
const fs = require('fs');
const path = require('path');

// Path to the Excel file
const excelFilePath = path.join(__dirname, 'Other', 'visiondb.xlsx');

// Path for the output CSV file
const csvFilePath = path.join(__dirname, 'Other', 'visiondb.csv');

try {
  console.log(`Reading Excel file: ${excelFilePath}`);
  
  // Read the Excel file
  const workbook = xlsx.readFile(excelFilePath);
  
  // Get the first sheet
  const sheetName = workbook.SheetNames[0];
  const worksheet = workbook.Sheets[sheetName];
  
  // Convert to CSV
  console.log(`Converting to CSV: ${csvFilePath}`);
  const csv = xlsx.utils.sheet_to_csv(worksheet);
  
  // Write to file
  fs.writeFileSync(csvFilePath, csv);
  
  console.log(`Conversion complete! CSV file saved to: ${csvFilePath}`);
  
  // Get some info about the data
  const jsonData = xlsx.utils.sheet_to_json(worksheet);
  console.log(`Number of rows: ${jsonData.length}`);
  if (jsonData.length > 0) {
    console.log(`Columns: ${Object.keys(jsonData[0]).join(', ')}`);
  }
  
} catch (error) {
  console.error(`Error converting file: ${error.message}`);
}
